pkg/proxy/ipvs/proxier.go

```diff
@@ -711,6 +711,11 @@ func CanUseIPVSProxier(handle KernelHandler, ipsetver IPSetVersioner) (bool, err
 	}
 	mods = utilipvs.GetRequiredIPVSModules(kernelVersion)
 	wantModules := sets.NewString()
+	schedulerMod, modNotFound := utilipvs.GetRequiredSchedulerModules(scheduler)
```

Comment:
I think `GetRequiredSchedulerModules` should check the list of mods installed on the host and default to RR if the provided scheduler doesn't match any of the installled modules. I think we can check loaded mods from ` loadModules` above.

Reply:
Currently I am defaulting to the RR scheduler [here](https://github.com/kubernetes/kubernetes/pull/93040/files#diff-8158c9c30c9a7cdffcc91ccf18590e11R170-R173) unless something is passed.

I think I originally added the check for all but the [feedback](https://github.com/kubernetes/kubernetes/pull/93040#discussion_r454477459) was to switch to interpolating only if we pass the flag which will short circuit looking for any of the other modules.

Reply:
I see, so my comment here https://github.com/kubernetes/kubernetes/pull/93040#discussion_r456143914 said:

> For any non-standard scheduler, I think we should interpolate the module name (ip_vs_'scheduler') and log a warning if it's not installed with a fallback back to RR.

To me, this means we check for a module `ip_vs_<provided-scheduler>` and if that isn't in the list of loaded modules, we fall back to RR. This way we're not maintaining a list of known schedulers, we can just check what is already loaded in the kernel. What do you think?

Reply:
For context on why I added these to the list -  ip_vs_sh & ip_vs_wrr are currently listed as required modules as of https://github.com/kubernetes/kubernetes/pull/59566. 

As a side effect should we also drop the other 2 modules that are not inherently necessary for round robin ?

This list feels like a clean way to include current scheduler modes and may consequently allow us to feature gate future algorithms added to IPVS. I'm fine with dropping the static list if it is of no use.

Reply:
While looking at some tests, I just noticed that we keep a list of valid values for the scheduler configuration in the [proxy validation code](https://github.com/kubernetes/kubernetes/blob/e79f49ebba9ab7402ce1d195d6ada3f234862551/pkg/proxy/apis/config/validation/validation.go#L245-L256). Perhaps this means it's fine to just have the list there but I think my question abouve about sh & wrr are still valid.

Reply:
> As a side effect should we also drop the other 2 modules that are not inherently necessary for round robin ?

I think so, but need to think about it more. I don't think `ip_vs_sh` and `ip_vs_wrr` are strictly required, but are common enough and were originally listed as requirements. 

> While looking at some tests, I just noticed that we keep a list of valid values for the scheduler configuration in the proxy validation code. Perhaps this means it's fine to just have the list there but I think my question abouve about sh & wrr are still valid.

I don't think this is enforces what we pass into the flags though.

Reply:
> While looking at some tests, I just noticed that we keep a list of valid values for the scheduler configuration in the proxy validation code. Perhaps this means it's fine to just have the list there but I think my question abouve about sh & wrr are still valid.
> 
> I don't think this is enforces what we pass into the flags though.

That sounds right. I should be able to test this with flags even though the config is preferred right?